src/main/java/com/example/controllers/DashboardController.java

```diff
@@ -0,0 +1,206 @@
+	public String userDetail(final Model model, final Principal principal, @RequestParam(value = "username") final String username) {
+
+		Account account = accountDao.findUsersByUsername(username).get(0);
+		List<CreditAccount> creditAccounts = creditaccountDao.findCreditAccountsByUsername(principal.getName());
+
+		model.addAttribute("creditAccounts", creditAccounts);
+		model.addAttribute("account", account);
+		model.addAttribute("accountMalicious", account);
+
+		return "userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/avatar", method = RequestMethod.GET)
+	public void getImage(final HttpServletResponse response, @RequestParam(value = "image") final String image) {
+
+		File file = storageFacade.exists(image) ? storageFacade.load(image) : storageFacade.load("avatar.png");
+		try {
+			writeResponse(new FileInputStream(file), response.getOutputStream());
+		}
+		catch (FileNotFoundException e) {
+			e.printStackTrace();
+		}
+		catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@RequestMapping(value = "/userDetail/avatar/update", method = RequestMethod.POST)
+	public String updateAvatar(@RequestParam("imageFile") final MultipartFile imageFile, final Principal principal,
+			final RedirectAttributes redirectAttributes) {
+		if (!imageFile.isEmpty()) {
+			try {
+				storageFacade.save(imageFile.getInputStream(), principal.getName() + ".png"); 
+			}
+			catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+
+		redirectAttributes.addAttribute("username", principal.getName());
+
+		return "redirect:/dashboard/userDetail";
+	}
+
+	@RequestMapping(value = "/userDetail/certificate", method = RequestMethod.POST)
+	public void getCertificate(final HttpServletResponse response, final Account account) {
+
+		Account bdAccount = accountDao.findUsersByUsername(account.getUsername()).get(0);
+		File tmpFile = null;
+
+		try {
+			// Serialize object
+			tmpFile = File.createTempFile("serial", ".ser");
+			FileOutputStream fos = new FileOutputStream(tmpFile);
+			ObjectOutputStream oos = new ObjectOutputStream(fos);
+			oos.writeObject(new FileUntrustedValid(account.getName()));
+			oos.close();
+			fos.close();
+
+			// Write into response
+			response.setHeader("Content-Disposition", "attachment; filename=\"Certificate_" + bdAccount.getName() + "_" + ".jks\"");
+			int length = writeResponse(new FileInputStream(tmpFile), response.getOutputStream());
+			response.setContentType("application/octet-stream");
+			response.setHeader("Content-Length", length + "");
+
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@RequestMapping(value = "/userDetail/newcertificate", method = RequestMethod.POST)
+	@ResponseBody
+	public String processSimple(@RequestParam(value = "file", required = false) final MultipartFile file, final Model model)
+			throws IOException, ClassNotFoundException {
+
+		ObjectInputStream ois = new ObjectInputStream(file.getInputStream());
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 8d6b8715bd8a4bc99194ad5292ee9b1b -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Insecure Object Deserialization
_High CWE-502_
A user-controllable string is deserialized.

An attacker can instantiate arbitrary classes, possibly resulting in a denial of service or potentially unintended code execution.

### How to fix
The application should avoid using object deserialization for data shared outside a trusted system.

The following things should be taken into consideration for remediation of this issue:

* Sensitive data should never be included without in serialized state that is sent to a client. If sensitive data is included in serialized state, it must be encrypted with a key that is only available to authorized entities for decrypting the sensitive data. 
* Serialized data must be protected from unauthorized modification, i.e., it should use some form of integrity protection such as a keyed MAC or being digitally signed.
* Data entering via an untrusted source (e.g., client) should never be deserialized. If this is not possible, then the application must explicitly specify and enforce which classes that should be allowed to be deserialized. It is important to note that it is not sufficient to add validation checks within the class that the application expects to deserialize, as an attacker can create a serialized object of any class they choose, which would then bypass such validation. It is important to strictly control which classes are allowed to be deserialized.

If using Java, it is recommended to use a run-time "agent" such as notsoserial (https://github.com/kantega/notsoserial) to specify and control which classes that should be allowed to be deserialized. Furthermore, it is strongly recommended to use the latest hardened JDK and ensure all third-party libraries, such as Apache Commons Collections, are updated.